protocols/identify/src/protocol.rs

```diff
@@ -349,6 +275,13 @@ impl TryFrom<proto::Identify> for PushInfo {
     }
 }
 
+pub(crate) enum OperationOkValue {
+    SendIdentify,
+    ReceiveIdentify(Info),
+    SendIdentifyPush,
+    ReceiveIdentifyPush(PushInfo),
+}
```

Comment:
Instead of this, let's do the following:

```rust
pub(crate) enum Success<T> {
	Sent(T),
	Received(T)
}
```

Additionally, I'd prefer if the individual protocol functions don't know about this and just have the minimal signature required:

```rust
async fn send_identify(info: &Info) -> Result<(), UpgradeError> // Note how I changed the parameter to `&Info`.
```

In the connection handler, we can then do the following:

```rust
self.active_streams.push(async move {
	send_identify(&info, stream).await?;

	Ok(Success::Sent(info))
})

self.active_streams.push(receive_push(stream).map_ok(Success::Received));
```

The reasoning is that these functions shouldn't change for a requirement of a layer further up. Just because `ConnectionHandler` wants to poll all these functions in the same `FuturesSet` doesn't mean these functions should relax their type-signatures. In general, it is a more effective use of the type-system if a function only has minimal parameters (and thus dependencies).

Reply:
The enum I am suggesting above may seem a bit odd but it will help us for https://github.com/libp2p/rust-libp2p/pull/4527 where we are changing the return type anyway.

Reply:
It is needing to differ between `sent` for `identify` and `identify/push` messages. Isn't it?
If we have just 'Sent' enum how could we match?

Reply:
> If we have just 'Sent' enum how could we match?

Based on the embedded type: `Info` vs `PushInfo`!

Reply:
As I understand, `Success<T>` can't be used as a type for `FuturesSet`, it should have a concrete type.

Reply:
Ah yeah, of course, my bad! We'd have to have two `FuturesSet`s again then.